akka-actor/src/main/scala-2.11/akka/util/ByteIterator.scala

```diff
@@ -277,8 +277,8 @@ object ByteIterator {
 
     final override def takeWhile(p: Byte => Boolean): this.type = {
       var stop = false
-      var builder = new ListBuffer[ByteArrayIterator]
-      while (!stop && !iterators.isEmpty) {
+      val builder = new ListBuffer[ByteArrayIterator]
```

Comment:
Just copied over 2.12 version and applied the fixes necessary so they are as similar as possible.